### PR TITLE
Add theme colors to Storybook

### PR DESCRIPTION
**Pull request by pnavarrc**
Add theme colors to Storybook.

image

I just picked up colors automatically from the theme, and excluded `mode` which seems to be an internal MUI thing.

### CODE REVIEW

File: packages/ui-components/src/stories/Colors.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+import { Typography, Palette, Grid, Box } from "@mui/material";
+import theme from "../styles/theme";
+
+export default {
+  title: "Design System/Colors",
+};
+
+const paletteGroups = (Object.keys(theme.palette) as (keyof Palette)[])
+  .filter((name) => isObject(theme.palette[name]))
+  .filter((name) => ["mode"].includes(name))
```
Comment (mikelehen) [inline packages/ui-components/src/stories/Colors.stories.tsx]: Is this equivalent to `name === 'mode'`? Or am I confused?  If so, then we don't need any of this `Object.keys(...).filter(...)` stuff, right?  We could just basically do `const name = 'mode'` ?  Feel like I may be confused. 😁

**Review by mikelehen — CHANGES_REQUESTED** [review]
Hrm.  I'm not sure if this is working for me.   https://act-now-packages--pr80-pablo-colors-to-stor-hp3f4gs8.web.app/storybook/index.html?path=/story/design-system-colors--all is not rendering anything.  Am I confused?

File: packages/ui-components/src/stories/Colors.stories.tsx (hunk 1 above)
Comment (pnavarrc) [reply packages/ui-components/src/stories/Colors.stories.tsx]: That was a bug 😅  — Some of the palette attributes are functions or non-string constants (such as `getContrastText()` and `contrastThreshold`), so I'm filtering out those. The `mode` attribute is used to name themes (`light`, `dark`), so is also not a color, so I'm filtering it out.

Comment (pnavarrc) [reply packages/ui-components/src/stories/Colors.stories.tsx]: Adding a comment to explain that 👍

File: packages/ui-components/src/stories/Colors.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import theme from "../styles/theme";
+
+export default {
+  title: "Design System/Colors",
+};
+
+// Filter out the palette attributes that don't correspond to colors,
+// such as functions or numeric constants. The `mode` attribute is
+// a string to name themes (dark, light, highContrast), but not a color,
+// so we filter that out as well.
```
Comment (mikelehen) [inline packages/ui-components/src/stories/Colors.stories.tsx:13]: Very helpful comment, thanks!

**Review by mikelehen — APPROVED** [review]
Looks good!

**Pull request merged**